core/string/ucaps.h

```diff
@@ -1412,4 +1412,44 @@ static int _find_lower(int ch) {
 	return ch;
 }
 
+static bool is_unicode_upper_case(int ch) {
```

Comment:
Probably binary search should be replaced with ranges (as in `char_utils.h`), but I'm not familiar with the Unicode specification, so I don't know where to find them on the official site. But I found these lists:

* https://www.compart.com/en/unicode/category/Ll
* https://www.compart.com/en/unicode/category/Lu

Reply:
Official Unicode spec for casing is here:

Data files - https://unicode.org/Public/UNIDATA/
Info - https://www.unicode.org/reports/tr44/#Casemapping